pil-analyzer/src/statement_processor.rs

```diff
@@ -340,17 +340,19 @@ where
             PilStatement::ConnectIdentity(source, left, right) => (
                 source,
                 IdentityKind::Connect,
-                SelectedExpressions {
+                powdr_ast::parsed::SelectedExpressions {
                     selector: None,
-                    expressions: self
-                        .expression_processor(&Default::default())
-                        .process_expressions(left),
+                    expressions: Box::new(
+                        self.expression_processor(&Default::default())
+                            .process_vec_into_array_literal(left),
```

Comment:
Is there a way to handle connect in the same way as `in`? I know it doesn't have a selector but that's the only difference, right?